Persistence/Repositories/ArticleRepository.cs

```diff
@@ -15,23 +17,32 @@ public ArticleRepository(AppDbContext context) : base(context)
         {
         }
 
-        public async Task<IEnumerable<Article>> ListAsync()
+        public async Task<IEnumerable<Article>> ListAsync(ArticlesParameters queryParameters)
         {
-            return await _context.Articles
+            IQueryable<Article> articlesQuery = _context.Articles
                 .Include(a => a.Release)
                 .ThenInclude(r => r.ProductVersion)
-                .ThenInclude(pr => pr.Product)
-                .ToListAsync();
-        }
+                .ThenInclude(pr => pr.Product);
+            
+            if (queryParameters.product > 0)
+            {
+                articlesQuery = articlesQuery.Where(a => a.Release.ProductVersion.Product.Id == queryParameters.product);
+            }
 
-        public async Task<IEnumerable<Article>> ListByProductAsync(int pid)
-        {
-            return await _context.Articles
-                .Include(a => a.Release)
-                .ThenInclude(r => r.ProductVersion)
-                .ThenInclude(pr => pr.Product)
-                .Where(article => article.Release.ProductVersion.Product.Id == pid)
-                .ToListAsync();
+            if (!String.IsNullOrEmpty(queryParameters.sort))
+            {
+                switch (queryParameters.sort)
+                {
+                    case "old":
+                        //articlesQuery = articlesQuery.OrderByDescending(a => a.Date);
+                        break;
+                    case "new":
+                        //articlesQuery = articlesQuery.OrderBy(a => a.Date);
+                        break;
+                }
```

Comment:
Ka gjør den "tilstandsmaskinen" her?

Reply:
Velge korleis dataen skal sorterast. Koden e kommentert ut fordi article.Date ikkje e implementert enda

Reply:
Takk for svar, HG